src/main/java/net/dv8tion/jda/webhook/WebhookClientBuilder.java

```diff
@@ -81,9 +82,11 @@ public WebhookClientBuilder(@Nonnull String url)
         Matcher matcher = WEBHOOK_PATTERN.matcher(url);
         if(!matcher.find()) {
             throw new IllegalArgumentException("Failed to parse webhook URL");
+        } else if (!matcher.matches()) {
+            throw new IllegalArgumentException("Failed to match the entire string as a webhook URL");
         }
```

Comment:
You can just drop the `matcher.find()` part completely in favor of `matcher.matches()`

Reply:
Looking at the javadoc I believe find() is still required to get the group data

Reply:
<https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html#matches-->
`If the match succeeds then more information can be obtained via the start, end, and group methods.`

Reply:
Ah, I'm blind. Nvm

Reply:
Actually didn't find any indication about only `find()` populating groups when going over class docs and comparing `find()` vs `matches()` docs

Reply:
Just ran a test. Works fine with only `matches()`